project/src/test/java/com/google/sps/data/PlacesTest.java

```diff
@@ -54,4 +39,77 @@ public void randomSort_keepsAllItems() {
     assertTrue(result.contains(PLACE_1));
     assertTrue(result.contains(PLACE_2));
   }
+
+  @Test
+  public void filter_noNeedToFilter_noFilter() {
+    ImmutableList<Place> twoPlaces = ImmutableList.of(PLACE_1, PLACE_2);
+
+    ImmutableList<Place> result = Places.filter(
+      twoPlaces /* places */,
+        1 /* min rating */,
+        true /* filter if no website */,
+        true /* filter branches of same place */
+    );
+
+    assertEquals(result, twoPlaces);
+  }
+
+  @Test
+  public void filter_fewBranchesOfSamePlace_filterOutBranches() {
+    Place branch1 = getValidPlaceBuilder().setLocation(new LatLng(35.35, 30.30)).build();
+    Place branch2 = getValidPlaceBuilder().setLocation(new LatLng(30.30, 30.30)).build();
+    Place branch3 = getValidPlaceBuilder().setLocation(new LatLng(35.35, 35.35)).build();
+
+    ImmutableList<Place> result = Places.filter(
+        ImmutableList.of(branch1, branch2, branch3) /* places */,
+        1 /* min rating */,
+        false /* filter if no website */,
+        true /* filter branches of same place */
+      );
+
+    assertEquals(result, ImmutableList.of(branch1));
+  }
+
+
+  @Test
+  public void filter_noWebsite_filterOut() {
+    Place websiteEmpty = getValidPlaceBuilder().setWebsiteUrl("").build();
+
+    ImmutableList<Place> result = Places.filter(
+        ImmutableList.of(websiteEmpty, PLACE_1, PLACE_2) /* places */,
+        1 /* min rating */,
+        true /* filter if no website */,
+        false /* filter branches of same place */
+    );
+
+    assertEquals(result, ImmutableList.of(PLACE_1, PLACE_2));
+  }
+
+  @Test
+  public void filter_tooLowRating_filterOut() {
+    int highRating = 5;
+    int lowerRating = 4;
+    Place highRatingPlace = getValidPlaceBuilder().setRating(highRating).build();
+    Place lowRatingPlace = getValidPlaceBuilder().setRating(lowerRating).build();
+
+    ImmutableList<Place> result = Places.filter(
+        ImmutableList.of(highRatingPlace, lowRatingPlace) /* places */,
+        highRating /* min rating */,
+        false /* filter if no website */,
+        false /* filter branches of same place */
+    );
+
+    assertEquals(result, ImmutableList.of(highRatingPlace));
+  }
+
+  // Returns a Place builder that has valid values of all attributes.
+  private static Place.Builder getValidPlaceBuilder() {
```

Comment:
Another comment regarding this method is that the only thing that matters here is the name (for test's sake). I'd make it a method argument so it's clear as sunshine that you're creating 3 valid places with the same name (for example).

You can name this createValidPlaceByName and return an instance instead of a builder.

If you want to change an argument during the test to distinguish the objects, you can do so by adding a toBuilder() method to Place class (which will return a Builder object of that instance), so for example in filter_fewBranchesOfSamePlace_filterOutBranches you can do:

ImmutableList.of(createValidPlaceByName("place 1").toBuilder().setRating(1).build(), createValidPlaceByName("place 2").toBuilder().setRating(2).build(), createValidPlaceByName("place 3").toBuilder().setRating(3).build());

Reply:
Great ideas, thanks. done